### ISSUE TITLE
Error on sparql query

### ISSUE THREAD

**Issue by AlexSerrano22** [CarbonLDP/carbonldp-js-sdk#221]
I'm working with documents.sparql to obtain the name of my projects as the first step to create a query more complex but I'm getting an error:

this is my code: 
```
carbon.documents.sparql( "projects/" )
      .prefix( "ldp", "http://www.w3.org/ns/ldp#" )
      .select( "projectName" )
      .where( _ => {
        return [
          _.resource( "projects/" )
            .has( "ldp:contains", _.var( "everyProject" ) ),
          _.var( "everyProject" )
            .has( "https://schema.org/name", _.var( "projectName" ) )
        ];
      } )
      .execute()
      .then( ( [ result, response ] ) => {
        return result
          .bindings
          .map( binding => {
              return { name: binding[ "projectName" ]}
          })
      } )
```

and the error is this: 
<img width="1425" alt="screen shot 2018-01-28 at 10 37 43 pm" src="https://user-images.githubusercontent.com/5456395/35493851-006841c8-047c-11e8-9b83-2d58078e9ad9.png">

I tried to change for another property that has a custom prefix:

```
carbon.documents.sparql( "projects/" )
      .prefix( "ldp", "http://www.w3.org/ns/ldp#" )
      .prefix( "tms", "https://tms.base22.com/ns#" )
      .select( "projectForecastID" )
      .where( _ => {
        return [
          _.resource( "projects/" )
            .has( "ldp:contains", _.var( "everyProject" ) ),
          _.var( "everyProject" )
            .has( "tms:forecastID", _.var( "projectForecastID" ) )
        ];
      } )
      .limit( 10 )
      .execute()
      .then( ( [ result, response ] ) => {
        return result
          .bindings
          .map( binding => {
              return { forecastID: binding[ "projectForecastID" ]}
          })
      } )
```
this code do a request but return with 400 error, the query that was executed is this:
```
BASE<http://carbonldp.webtraining.zone:8083/>PREFIX tms:<https://tms.base22.com/ns#>PREFIX ldp:<https://www.w3.org/ns/ldp#>PREFIX ldp:<http://www.w3.org/ns/ldp#>PREFIX tms:<https://tms.base22.com/ns#>SELECT?projectForecastID{</projects/>ldp:contains?everyProject.?everyProject tms:forecastID?projectForecastID}LIMIT10
```

this is the correct way to execute a sparql query whit the query builder? or I'm missing something?

**Comment by roddolf** [comment]
The first example is falling because a string parameter in the `.has()` method if for a RAW SPARQL path, so if you want to use a IRI you must use `<https://schema.org/name>`, or use a helper like `_.resource( "https://schema.org/name" )`

**Comment by roddolf** [comment]
The second one it appears the prefixes are been duplicated.
Do you have any of the prefix you declared by `.prefix()` in your general schema?
If thats the case, can you try to the delete the declaration by `.prefix()` an see what happen?

**Comment by roddolf** [comment]
Oh, I may not be too clear, I has taking about of the carbonldp schemas:
- `c`: `https://carbonldp.com/ns/v1/platform#`
- `cs`: `https://carbonldp.com/ns/v1/security#`

**Comment by AlexSerrano22** [comment]
thanks for your help

**Comment by roddolf** [comment]
I reopen the use since its still a bug that the prefix is not been replaced by the new declaration in `.prefix()`

**Comment by roddolf** [comment]
See CarbonLDP/sparqler#19

**Comment by roddolf** [comment]
Fixed with `v1.0.0.alpha-10`
